framework/vulkan_sample.h

```diff
@@ -1044,6 +1087,11 @@ inline bool VulkanSample<bindingType>::prepare(const ApplicationOptions &options
 		}
 	}
 
+#ifdef VK_ENABLE_BETA_EXTENSIONS

```

Comment:
Shouldn't this be guarded by some mac/ios specific define instead? It's possible that we enable beta extensions for future samples (afair we even did that at some point, maybe with ray tracing) and I guess this would fail for all samples then as it'll always request portability.

Reply:
@SaschaWillems I am away for a few days but should be able to respond tomorrow night EDT.

Reply:
@asuessenbach had exactly the same question.  This was my answer: https://github.com/KhronosGroup/Vulkan-Samples/pull/1084#discussion_r1669430006.  It's not a macOS/iOS thing, it's a _portability_ library thing, and I was trying to stay within the conventions of the project (see _portability_ sample which I have not touched).

However, given that @SaschaWillems and @asuessenbach have raised this, there is likely a semantics problem with using `VK_ENABLE_BETA_EXTENSIONS` as a dual-purpose thing within the project: i.e. to enable a) portabiliy and b) Vulkan Beta extensions (which include portability).  Perhaps we should instead be looking at expanding the meaning and use of `VKB_ENABLE_PORTABILITY` (i.e. to include portability enumeration _and_ features).  We would then set `VK_ENABLE_BETA_EXTENSIONS` as a dependency of `VKB_ENABLE_PORTABILITY` so the Vulkan library portability features are exposed, but **not** use it as a define within the project.   Would you like me to push an additional commit that makes this change?

Reply:
I looked at the code again, and found a way to use the existing `VKB_ENABLE_PORTABILITY` define to handle both the portability enumeration _and_ portability subset extensions in the framework.  The result is actually a bit better than before, with dynamic determination of `VK_KHR_portability_enumeration` in the runtime instance vs. reliance on static enablement in cmake config files.  This is important not just for support of older SDKs, but also to allow future support of MoltenVK standalone.  Note that `VK_KHR_portability_enumeration` is a Vulkan loader feature only, and not present in the actual driver.  When MoltenVK standalone is used, the extension is not present and the code needs to be flexible enough to handle this case.

This approach also solves the concerns of @SaschaWillems regarding possible future enablement of Vulkan beta extensions (`VK_ENABLE_BETA_EXTENSIONS`) for purposes other than portability.

The changes are in commit https://github.com/KhronosGroup/Vulkan-Samples/pull/1084/commits/6e73f7c60eba57b9dab968cd5134d6e14cebc9f6.  Please review and let me know whether this helps from your perspective.  If this does not suit your plans, then it could always be reverted.

Reply:
I am pretty sure this change is correct, so I am going to close this specific item and request a formal review later tonight.